apps/passport-server/src/services/devconnect/organizerSync.ts

```diff
@@ -787,12 +868,47 @@ export class OrganizerSync {
           await softDeleteDevconnectPretixTicket(this.db, removedTicket);
         }
 
-        span?.setAttribute("ticketsInserted", newTickets.length);
-        span?.setAttribute("ticketsUpdated", updatedTickets.length);
-        span?.setAttribute("ticketsDeleted", removedTickets.length);
+        let redactedTicketsDeleted = 0;
+        if (this.enableRedaction) {
+          // Step 4 of saving: save redacted tickets
+          // Grouping by position ID is safe here because all of these tickets
+          // belong to the same event, and position ID is locally unique.
+          const redactedTickets = await Promise.all(
+            ticketsToRedact.map(this.prepareRedactedTicket)
+          );
+
+          const redactionChanges =
+            compareArrays<DevconnectPretixRedactedTicket>(
+              redactedTicketsInDB,
+              redactedTickets,
+              "position_id",
+              pretixRedactedTicketsDifferent
+            );
+
+          for (const redactedTicket of [
+            ...redactionChanges.new,
+            ...redactionChanges.updated
+          ]) {
+            await upsertDevconnectPretixRedactedTicket(this.db, redactedTicket);
+          }
+
+          await deleteDevconnectPretixRedactedTickets(
+            this.db,
+            eventConfigID,
+            redactionChanges.removed.map((t) => t.position_id)
+          );
+
+          redactedTicketsDeleted = redactionChanges.removed.length;
+        }
+
+        span?.setAttribute("ticketsInserted", changes.new.length);
+        span?.setAttribute("ticketsUpdated", changes.updated.length);
+        span?.setAttribute("ticketsDeleted", changes.removed.length);
+        span?.setAttribute("redactedTicketsInserted", ticketsToRedact.length);
+        span?.setAttribute("redactedTicketsDeleted", redactedTicketsDeleted);
```

Comment:
you should probably handle tracing these the same way as the three lines above, right? i think `redactedTicketsInserted` the way you have it reports an incorrect value, it should be something like `redactionChanges.new.length`. and you don't have an entry for updated.